### PR TITLE
JSON:API collections have 'item' links to their members.

### PR DESCRIPTION
**Pull request by evert**
Relates to #109 

cc @gabesullice @ethanresnick

### CODE REVIEW

File: src/representor/jsonapi.ts [hunk 1]
```diff
@@ -8,15 +8,35 @@ import Representation from './base';
  */
 type JsonApiLink = string | { href: string };
 
+/**
+ * This type is a full 'links' object, which might appear on the top level
+ * or on resource objects.
+ */
+type JsonApiLinksObject = {
+  self?: JsonApiLink,
+  profile?: JsonApiLink
```
Comment (gabesullice) [inline src/representor/jsonapi.ts]: I would remove `profile`. It doesn't serve a purpose ATM.

File: src/representor/jsonapi.ts [hunk 2]
```diff
@@ -8,15 +8,35 @@ import Representation from './base';
...
+  [rel: string]: JsonApiLink | JsonApiLink[]
+};
+
+/**
+ * This is a single JSON:API resource. Its type contains just the properties
+ * we care about.
+ */
+type JsonApiResource = {
```
Comment (gabesullice) [inline src/representor/jsonapi.ts]: Looks good.

File: src/representor/jsonapi.ts [hunk 3]
```diff
@@ -8,15 +8,35 @@ import Representation from './base';
...
+  links?: JsonApiLinksObject,
+};
+
+
 /**
  * This type represents a valid JSON:API response. We're only interested
  * in the links object at the moment, so everything else is (for now)
  * untyped.
  */
 type JsonApiObject = {
```
Comment (gabesullice) [inline src/representor/jsonapi.ts]: This should be renamed to `TopLevelObject`. Sorry, I should have spotted that before. There's already a [JSON:API Object](https://jsonapi.org/format/#document-jsonapi-object) that looks different than this.

File: src/representor/jsonapi.ts [hunk 4]
```diff
@@ -71,6 +94,39 @@ function parseJsonApiLinks(baseHref: string, body: JsonApiObject): Link[] {
 
 }
 
+/**
+ * Find collection members in JSON:API objects.
+ *
+ * A JSON:API object might be a collection that has several members.
```
Comment (gabesullice) [inline src/representor/jsonapi.ts]: s/JSON:API/top-level

File: src/representor/jsonapi.ts (hunk 3 above)
Comment (evert) [reply src/representor/jsonapi.ts]: Fixed

**Pull request merged**